webpack/webpack.config.js

```diff
@@ -0,0 +1,78 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+// @ts-check
+
+"use strict";
+
+const path = require("path");
+const webpack = require("webpack");
+
+/**@type {import('webpack').Configuration}*/
+const config = {
+  //   stats: "minimal",
+  mode: "development",
+  target: "node", // vscode extensions run in a Node.js-context 📖 -> https://webpack.js.org/configuration/node/
+
+  entry: "./src/extension.ts", // the entry point of this extension, 📖 -> https://webpack.js.org/configuration/entry-context/
+  output: {
+    // the bundle is stored in the 'dist' folder (check package.json), 📖 -> https://webpack.js.org/configuration/output/
+    path: path.resolve(__dirname, "../out/prod"),
+    filename: "extension.js",
+    libraryTarget: "commonjs2",
+    devtoolModuleFilenameTemplate: "../../[resource-path]"
+  },
+  devtool: "source-map",
+  externals: {
+    vscode: "commonjs vscode", // the vscode-module is created on-the-fly and must be excluded. Add other modules that cannot be webpack'ed, 📖 -> https://webpack.js.org/configuration/externals/
+    electron: "electron" // the vscode-module is created on-the-fly and must be excluded. Add other modules that cannot be webpack'ed, 📖 -> https://webpack.js.org/configuration/externals/
+  },
+  resolve: {
+    // support reading TypeScript and JavaScript files, 📖 -> https://github.com/TypeStrong/ts-loader
+    extensions: [".ts", ".js", ".json"],
+    alias: {
+      "@abstractions": path.join(__dirname, "../src/abstractions/node")
```

Comment:
How does this work? In the future, we can require “@abstractions/*” and have it use two different implementations between node and the browser?

Reply:
Yep exactly, we will have to have an abstraction over a class and two implementations for the web and nodejs.